pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,188 @@
+            cvi.delete()
+
+
+def delete_repository_version_indexes_by_repository(repository):
+    """Delete indexes for a repository."""
+    for cvi in CVIndex.objects.filter(repository__pk=repository.pk):
+        cvi.delete()
+
+
+def update_index_for_repo_version(repository_version, is_latest):
+    """Rebuild index by repository."""
+
+    repository = repository_version.repository
+    repository.cast()
+
+    if is_latest:
+        repo_v = None
+    else:
+        repo_v = repository_version
+
+    if not _is_indexable(repository, repo_v):
+        return
+
+    cvs = repository_version.content.filter(pulp_type="ansible.collection_version")
+    cvs = [x.cast() for x in cvs]
+
+    # get the set of signatures in this repo version
+    repo_signatures = repository_version.content.filter(pulp_type="ansible.collection_signature")
+    repo_signatures = [x.cast() for x in repo_signatures]
+
+    # get the set of deprecations in this repo version
+    deprecations = repository_version.content.filter(pulp_type="ansible.collection_deprecation")
+    deprecations = [x.cast() for x in deprecations]
+    deprecations_set = set([(x.namespace, x.name) for x in deprecations])
+
+    namespaces = repository_version.content.filter(pulp_type="ansible.namespace")
+    namespaces = [x.cast() for x in namespaces]
+    namespaces = {x.name: x for x in namespaces}
+
+    # map out the namespace(s).name(s) for everything in the repo version
+    colset = set([(x.namespace, x.name) for x in cvs])
+
+    log.info("Repository collections ...")
+    log.info(colset)
+
+    # clean up any repo+namespace+name indexes that no longer live in the repo ...
+    cvindexes = CVIndex.objects.filter(repository=repository, repository_version=repo_v)
+    for cvi in cvindexes:
+        if (cvi.collection_version.namespace, cvi.collection_version.name) not in colset:
+            log.info(
+                (
+                    f"DELETE {cvi.repository.name}"
+                    + f".{cvi.collection_version.namespace}"
+                    + f".{cvi.collection_version.name}"
+                    + f".{cvi.collection_version.version} FROM INDEX"
+                )
+            )
+            cvi.delete()
+            continue
+        log.info(
+            (
+                f"KEEP {cvi.repository.name}"
+                + f".{cvi.collection_version.namespace}"
+                + f".{cvi.collection_version.name}"
+                + f".{cvi.collection_version.version} IN INDEX"
+            )
+        )
+
+    # iterate through each collection in the repository
+    for colkey in colset:
+        namespace = colkey[0]
+        name = colkey[1]
+
+        # get all the versions for this collection
+        related_cvs = [x for x in cvs if x.namespace == namespace and x.name == name]
+
+        # compute the highest version (allow pre-releases too?)
+        versions = [x.version for x in related_cvs]
```

Comment:
`semantic_version` should rate normal releases higher than pre-releases.

Reply:
```#!/usr/bin/env python

from semantic_version import Version

versions = ['1.0.0', '2.0.0', '3.0.0-rc.1']
versions = [Version(x) for x in versions]
print(max(versions))
```

Will print out the 3.0.0-rc.1 prerelease instead of 2.0.0

Reply:
Hmm, probably need something like this then:

```python
release_versions = [x for x in versions if not x.prerelease]
# Account for scenario where there are only pre-releases
highest_version = release_versions[0] if release_versions else versions[0]
```